contracts/src/PBHEntryPointImplV1.sol

```diff
@@ -251,16 +251,17 @@ contract PBHEntryPointImplV1 is IPBHEntryPoint, WorldIDImpl, ReentrancyGuardTran
         nonReentrant
         returns (IMulticall3.Result[] memory returnData)
     {
+        uint256 gasBefore = gasleft();
         uint256 signalHash = abi.encode(msg.sender, calls).hashToField();
         verifyPbh(signalHash, pbhPayload);
         nullifierHashes[pbhPayload.nullifierHash] = true;
 
-        returnData = IMulticall3(_multicall3).aggregate3{gas: pbhGasLimit}(calls);
+        returnData = IMulticall3(_multicall3).aggregate3(calls);
         emit PBH(msg.sender, signalHash, pbhPayload);
 
         // Check if pbh gas limit is exceeded
-        if (gasleft() > pbhGasLimit) {
-            revert GasLimitExceeded(gasleft());
+        if (gasBefore - gasleft() > pbhGasLimit) {
+            revert GasLimitExceeded(gasBefore - gasleft());
         }
 
         return returnData;
```

Comment:
We dont need to explicitly return `returnData` if its already specified via `returns (IMulticall3.Result[] memory returnData)`.